### DISCUSSION TITLE
Misorientation histogram plotting problems

### DISCUSSION THREAD

**Discussion by orlandinebsd** [mtex-toolbox/mtex#1206] [Ask Anything] [answered]
Hello again all... sorry to be blowing up the forum like this, but hopefully this is a legit helpful question to ask. Still in the realm of misorientation analysis, but not MDFs like my other post so that each can be resolved clearly. 

So, one of my goals in misorientation analysis is to be able to compare histogram distributions of misorientation angles of grains that are neighbors, and misorientation angles of grains that are not neighbors. MDFs are one solution to this, but let's say that I want to stick with plain old histograms. I will use a data set with 2,022 quartz grains.

It turns out that getting the first value, the misorientation between grains that are neighbors, is pretty easy following this tutorial page [here](https://mtex-toolbox.github.io/GrainNeighbours.html). 

```
pairs = grains('quartz').neighbors;
grain_neighbor_mori = inv(grains(pairs(:,1)).meanOrientation) .* grains(pairs(:,2)).meanOrientation

grain_neighbor_mori = misorientation (show methods, plot)
  size: 978 x 1
  crystal symmetry : Quartz-new (321, X||a*, Y||b, Z||c*)
  crystal symmetry : Quartz-new (321, X||a*, Y||b, Z||c*)
```
Great, now I have 978 misorientations between all neighboring grain pairs. That number makes some sense based on the distribution of quartz grains in this particular map.

The second value, misorientations between randomly selected grain mean orientations, is not quite as straightforward but by using [this ](https://github.com/mtex-toolbox/mtex/discussions/1198)discussion I think it can be kludged as so:  

```
ebsdS('quartz').orientations = grains(ebsdS('quartz').grainId).meanOrientation;
grain_random_pairs_mori = calcMisorientation(ebsdS('quartz'),ebsdS('quartz'));

grain_random_pairs_mori = misorientation (show methods, plot)
  size: 98140 x 1
  crystal symmetry : Quartz-new (321, X||a*, Y||b, Z||c*)
  crystal symmetry : Quartz-new (321, X||a*, Y||b, Z||c*)
```
This kludge is necessary because 'calcMisorientation' refuses to work with anything but 'ebsd' data and I don't trust myself enough to fool around with 'inv' and 'angle' competently. The number of orientations calculated also kinda makes some intuitive sense because the spatial distribution of grains doesn't matter, and theoretically it is every grain against every other grain so it should be more than the total number of grains (although... shouldn't it be 2,043,231 possible unique pairs from 2,022 grains?). 

SO, now we have two misorientation objects that probably are what I think they are. I can theoretically plot this via 'plotAngleDistribution' or 'histogram': 

**plotAngleDistribution version**

```
figure
plotAngleDistribution(grain_random_pairs_mori,'binwidth',2,'DisplayName','Random grainmeans')
hold on
plotAngleDistribution(grain_neighbor_mori,'binwidth',2,'DisplayName','HAGB segments')
plotAngleDistribution(odf_grains.CS,odf_grains.CS,'displayName','Random','linewidth',2,'Color','black')

hold off
legend('Location','northeast') % update legend
set(gca, 'FontName', 'Times New Roman')
set(gca, 'FontSize', 16)
ylim([0 15])
xlabel('HAGB Misorientation Angle')
```
![BH1E HAGB Misorientation Angle via plotAngleDistribution](https://user-images.githubusercontent.com/90639180/133333246-a4cce4ea-2687-4c4d-9079-00b84c949014.png)

**histogram version**

```
figure
histogram(grain_random_pairs_mori.angle./degree,'binwidth',2,'Normalization','probability','DisplayName','Random grainmeans')
hold on
histogram(grain_neighbor_mori.angle./degree,'binwidth',2,'Normalization','probability','DisplayName','Neighbor grainmeans')
hold off
legend('Location','northeast') % update legend
set(gca, 'FontName', 'Times New Roman')
set(gca, 'FontSize', 16)
ylim([0 0.15])
xlabel('HAGB Misorientation Angle')
```

![BH1E HAGB Misorientation Angle via histogram](https://user-images.githubusercontent.com/90639180/133332167-a3df0c74-bcc1-4d6e-af0f-d7b7903b400f.png)

Both have big Dauphine twin spikes around 60 degrees in the neighbor dataset, which makes sense for the sample and suggests to me that these values are in fact what I think they are (interesting that there is a bit of a ramp-up from lower angles, followed by a sharp cutoff past the threshold, nay?). The twins aren't removed here because as has been pointed out many times on this forum the grain.meanOrientation is nonsense following a twin boundary merge. However, a few issues:
- Why are the frequency values different?
- Why does the plotAngleDistribution version ignore my attempt to use 'DisplayName'? It also ignores my attempts to use different colors, unless I edit the colors in the MATLAB figure window, as well as my attempts to define the bins. This function does play nicely when dealing with MDFs, such as with the 'Random' distribution plotted - it just gets obstinate when making histograms, apparently. 

The MATLAB histogram looks nicer, but only because the MTEX version refuses to let me modify any of the features. The MTEX version would be preferred because I assume the difference in frequency is due to some arcane, 100% correct consideration implemented by the geniuses who work on this program, and because superimposing the 'random' texture is always pretty helpful to me.

So. What gives with 'plotAngleDistribution' when it comes to histograms? Also, did I even plot the things that I thought I was plotting?

**Comment by kilir** [comment, marked as answer]
Hi,
with respect to the differences between the histogram populations: I would say it's related to the different binning you use. 
You are right in your observation that the DisplayName argument isn't considered, for now, a workaround is for example:
```
p = plotAngleDistribution(mori)
legend
p.DisplayName =  'somethingelse';
```
Cheers,
Rüdiger

**Comment by orlandinebsd** [comment]
Hello Rüdiger! 

Yes, you were correct about the bin sizes. I modified the default bin size from 5 degrees to 2 degrees on line 67 of 'plotAngleDistribution.m' in mtex-5.7.0\geometry to match what I was doing with 'histogram' and the values are now the same. 

This is a cool trick! I can't quite get it to work with a figure that also requires the use of 'hold' to plot the second set of bars, but I will try to figure it out. It would be great to use this to change the default colors from red and orange, too. If you get a chance, could you point me toward the general MATLAB functionality you are using to do this? Thank you so much for all of your help, Rüdiger! 

Best, Phil

**Comment by zmichels** [comment]
Personally, I prefer to use the Matlab-native histogram() function to generate my plots with more in-line control over binning, coloring, normalization, scaling/axes, legend, etc. https://www.mathworks.com/help/matlab/ref/matlab.graphics.chart.primitive.histogram.html     From: orlandinebsdSent: Wednesday, September 15, 2021 9:16 AMTo: mtex-toolbox/mtexCc: SubscribedSubject: Re: [mtex-toolbox/mtex] Misorientation histogram plotting problems (Discussion #1206) Hello Rüdiger!Yes, you were correct about the bin sizes. I modified the default bin size from 5 degrees to 2 degrees on line 67 of 'plotAngleDistribution.m' in mtex-5.7.0\geometry to match what I was doing with 'histogram' and the values are now the same.This is a cool trick! I can't quite get it to work with a figure that also requires the use of 'hold' to plot the second set of bars, but I will try to figure it out. It would be great to use this to change the default colors from red and orange, too. If you get a chance, could you point me toward the general MATLAB functionality you are using to do this? Thank you so much for all of your help, Rüdiger!Best, Phil—You are receiving this because you are subscribed to this thread.Reply to this email directly, view it on GitHub, or unsubscribe.

**Reply by DriptD** [reply]
Hello Dr. Michels,

Could you please let me know how to overlay the theoretical distribution curve on the correlated and uncorrelated misorientation angle distributions plotted using Matlab's default histogram function? Thanks!

Sincerely,
Dripta

**Comment by lfgmorales** [comment]
Hi Dripta

Why not use what is available for mtex?

<https://mtex-toolbox.github.io/HomepageOld/files/doc/GrainMisorientationAnalysis.html>
Misorientation Analysis (DocHelp Toolbox)<https://mtex-toolbox.github.io/HomepageOld/files/doc/GrainMisorientationAnalysis.html>
mtex-toolbox.github.io<https://mtex-toolbox.github.io/HomepageOld/files/doc/GrainMisorientationAnalysis.html>
        [X] <https://mtex-toolbox.github.io/HomepageOld/files/doc/GrainMisorientationAnalysis.html>



Cheers


L

Luiz F. G. Morales
ScopeM - ETH Zürich
Sent from iPhone

On 23 Mar 2023, at 11:30, Dripta Dutta ***@***.***> wrote:

﻿

Hello Dr. Michels,

Could you please let me know how to overlay the theoretical distribution curve on the correlated and uncorrelated misorientation angle distributions plotted using Matlab's default histogram function? Thanks!

Sincerely,
Dripta

—
Reply to this email directly, view it on GitHub<https://github.com/mtex-toolbox/mtex/discussions/1206#discussioncomment-5403812>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ALE5IN7XZORGKFDUH2KKND3W5QQ4PANCNFSM5EBAPX3A>.
You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

[ { ***@***.***": "http://schema.org", ***@***.***": "EmailMessage", "potentialAction": { ***@***.***": "ViewAction", "target": "https://github.com/mtex-toolbox/mtex/discussions/1206#discussioncomment-5403812", "url": "https://github.com/mtex-toolbox/mtex/discussions/1206#discussioncomment-5403812", "name": "View Discussion" }, "description": "View this Discussion on GitHub", "publisher": { ***@***.***": "Organization", "name": "GitHub", "url": "https://github.com" } } ]

**Reply by DriptD** [reply]
Hi Dr. Morales,

I want to overlay the uncorrelated histogram on the correlated one (shown below). But MTEX's plotAngleDistribution() function places the bars beside one another, thinning them, when I use 'hold on'. Therefore, I used Matlab's histogram() but now I cannot plot the theoretical curve. Does plotAngleDistribution() allow overlaying?

![image](https://user-images.githubusercontent.com/77139598/227179746-904012c2-84a7-434a-b6a3-933f38e89a6b.png)

Sincerely,
Dripta
